src/pubsub/mod_pubsub.erl

```diff
@@ -4470,7 +4470,7 @@ string_to_ljid(JID) ->
 
 -spec uniqid() -> mod_pubsub:itemId().
 uniqid() ->
-    {T1, T2, T3} = timestamp(),
+    {T1, T2, T3} = os:timestamp(),
     iolist_to_binary(io_lib:fwrite("~.16B~.16B~.16B", [T1, T2, T3])).
```

Comment:
Maybe we could put some kind of a UUID here or just use the integer timestamp? I think that the triplet is not even guaranteed to be unique, as it is the raw OS timestamp - not monotonic, right?

Reply:
Excellent idea, there's a tiny chance the os will warp at the same time two of these are generated and they might end up generating the same id. And this doesn't require any time knowledge, it's nothing more than a unique binary. Moving it to uuid_v4 :)